File: Strings/242-valid-anagram/valid-anagram.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean isAnagram(String s, String t) 
    {
        Map<Character, Integer> dict1 = new HashMap<>();
        Map<Character, Integer> dict2 = new HashMap<>();
        char c;

        for(int i=0; i<s.length(); i++)
        {
            c = s.charAt(i);
            if(!dict1.containsKey(c))
            {
                dict1.put(c,1);
            }
            else
            {
                int value = dict1.get(c);
                value = value+1;
                dict1.put(c, value);
            }
        }

        for(int i=0; i<t.length(); i++)
        {
            c = t.charAt(i);
            if(!dict2.containsKey(c))
            {
                dict2.put(c,1);
            }
            else
            {
                int value = dict2.get(c);
                value = value+1;
                dict2.put(c, value);
            }
        }

        boolean areEqual = dict1.equals(dict2);

        return areEqual;
    }
}

File: Strings/242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        l1 = [0] * 128
        l2 = [0] * 128

        for i in s:
            value = ord(i)
            l1[value] += 1


        for i in t:
            value = ord(i)
            l2[value] += 1

        for i in range(len(l1)):
            if l1[i] != l2[i]:
                return False

        return True
